libfreerdp/codec/planar.c

```diff
@@ -958,6 +958,7 @@ BYTE* freerdp_bitmap_planar_compress_plane_rle(const BYTE* inPlane,
 	UINT32 outBufferSize;
 	UINT32 nBytesWritten;
 	UINT32 nTotalBytesWritten;
+	BOOL allocate = !outPlane;
```

Comment:
Why free that here? The buffers are allocated and kept in the planar context struct.

Reply:
The function seems to be a little bit unfortune. You can pass `outPlane`, which is modified and returned. But you can also pass `NULL` and `malloc` is used consequently (though not sure that this case is ever used). In case of failure, `NULL` is returned and pointer to newly allocated memory is lost, but it should be freed. Maybe the function should rather return `BOOL` and expect that `outPlane` is always non-`NULL`.

Reply:
@ondrejholy you're right, the whole allocation stuff should go. (`freerdp_bitmap_planar_context_reset` allocates the planes)